RIOT-OS/applications
# LwM2M/SAUL temperature app

### Contribution description
This application extends the RIOT LwM2M Wakaama example to periodically sample a SAUL temperature sensor and make the value available as a LwM2M measured temperature (/3303) object. The app also can be run with a "native" board without a SAUL sensor. In this case the temperature value simply increases with each measurement.

This implementation provides partial support for the LwM2M definition for [measured temperature](http://www.openmobilealliance.org/tech/profiles/lwm2m/3303.xml), including only the sensor value and units.

The app works fine, but should be considered a WIP for a couple of reasons:
*  The Wakaamaa pkg was merged after 2019.10, and we don't expect to update this project's dependency on RIOT until the 2020.01 release. At that time I can remove the commit in this PR for that dependency. 
* The `measured_temp.(c|h)` module belongs with the Wakaama pkg in RIOT. I plan to create a separate PR to add that. Once that is merged, I can update this PR to reference it instead.

### Testing procedure
See [README.md](https://github.com/kb2ma/riot-apps/blob/wakaama-temp/wakaama-temp/README.md) to set up and run the app.

For temperature readings, I use a Microchip MCP9808 sensor with the RIOT JC42 driver. Adafruit [makes it easy](https://www.adafruit.com/product/1782) for those of us who live in the higher levels of the stack. :-)
